### PR TITLE
Update icewind/smb to 2.0.0

### PR DESCRIPTION
**Pull request by icewind1991**
Supersedes #2528

- depend on stdbuf in less cases
- explicitly check if the smbclient binary is available
- misc code improvements
- changes to the notify code (backwards incompatible, hence the version bumb) that allow for more flexible usage

cc @MorrisJobke @rullzer @LukasReschke please review

### CODE REVIEW

**Comment by icewind1991** [comment]
(ci errors are timeouts and jsunit :see_no_evil: )

File: apps/files_external/3rdparty/icewind/smb/src/NativeState.php [hunk 1]
```diff
@@ -31,44 +22,39 @@ class NativeState {
 
 	protected $connected = false;
 
+	// todo replace with static once <5.6 support is dropped
```
Comment (LukasReschke) [inline apps/files_external/3rdparty/icewind/smb/src/NativeState.php]: Already the case for stable11 and master but your decision if you want to do the same for your library.

File: apps/files_external/3rdparty/icewind/smb/src/Parser.php [hunk 2]
```diff
@@ -9,70 +9,93 @@
...
 
 class Parser {
+	const MSG_NOT_FOUND = 'Error opening local file ';
+
 	/**
 	 * @var \Icewind\SMB\TimeZoneProvider
 	 */
 	protected $timeZoneProvider;
 
+	// todo replace with static once <5.6 support is dropped
```
Comment (LukasReschke) [inline apps/files_external/3rdparty/icewind/smb/src/Parser.php]: Same

**Comment by LukasReschke** [comment]
Code looks good but didn't test yet, so only left some remarks but may take a look later and test it.

**Comment by icewind1991** [comment]
@rullzer @LukasReschke *poke*

**Comment by LukasReschke** [comment]
I'd be more confident to merge this once we have the files_external SMB tests running on CI.

cc @rullzer

**Comment by LukasReschke** [comment]
Let's first get https://github.com/nextcloud/server/pull/2789 in

**Comment by icewind1991** [comment]
rebased and smb tests are all green

**Comment by MorrisJobke** [comment]
> Let's first get #2789 in

Done :)

**Review by MorrisJobke — APPROVED** [review]

**Comment by icewind1991** [comment]
@LukasReschke please review

**Comment by MorrisJobke** [comment]
> @LukasReschke please review

Ping

**Review by rullzer — APPROVED** [review]

**Pull request merged**